src/include/golioth_debug.h

```diff
@@ -4,6 +4,11 @@
 #include "golioth_time.h"
 #include <stdbool.h>
 
+#ifndef LOG_TAG_DEFINE
+#define LOG_TAG_DEFINE(tag)                     \
+    static const char *TAG = #tag
```

Comment:
Seems okay. Since this is a breaking change, let's go ahead and add it to the Changelog for the next release.